databuilder/__main__.py

```diff
@@ -27,6 +29,7 @@
     "databricks": "databuilder.backends.databricks.DatabricksBackend",
     "graphnet": "databuilder.backends.graphnet.GraphnetBackend",
     "tpp": "databuilder.backends.tpp.TPPBackend",
+    "expectations": "databuilder.backends.dummy.DummyBackend",
```

Comment:
Instinctively, I'd take a slightly different approach here. I'll describe it and leave it to you as to whether you think it's an improvement or not.

I'd avoid having `expectations` listed as a valid option in the help text. I feel like this is a temporary fudge to get existing-databuilder working in existing-jobrunner, and not something we want to expose in the public API.

And I don't think we need a new class in `databuilder.backends` to achieve this. Instead we could use a sentinel object:
```py
EXPECTATIONS_BACKEND_PLACEHOLDER = object()
```

And then a guard clause on `backend_from_id`:
```py
def backend_from_id(str_id):
    if str_id == "expectations":
        return EXPECTATIONS_BACKEND_PLACEHOLDER
```

And then the asserts are just:
```py
assert options.backend is not EXPECTATIONS_BACKEND_PLACEHOLDER
```

It's just as horrible a fudge as before, but a bit less leaky in my view.

Reply:
I think this is better than what I have. I started using `None` as a signal and then decided that was _awful_, so I ended up with a whole dummy backend. Didn't occur to me to use a unique object. Thanks.